rust-cache/action.yml

```diff
@@ -24,21 +24,38 @@ runs:
   - shell: bash
     env:
       SCCACHE_VERSION: v0.4.1
-      SCCACHE_FILE: >-
-        ${{ (runner.os == 'Linux' && 'sccache-v0.4.1-x86_64-unknown-linux-musl.tar.gz')
-        || (runner.os == 'macOS' && 'sccache-v0.4.1-x86_64-apple-darwin.tar.gz')
-        || (runner.os == 'Windows' && 'sccache-v0.4.1-x86_64-pc-windows-msvc.tar.gz') }}
-      SCCACHE_FILE_SHA256: >-
-        ${{ (runner.os == 'Linux' && 'f077d92ca86d71bc55aebeeb6e8dc557fef481446ccc82504aeedf1fe6e1f657')
-        || (runner.os == 'macOS' && 'a291f1d90c6b25726866f018ec6071fa4d20ca443ad91fe5dfb9740eb4ebc45a')
-        || (runner.os == 'Windows' && '7508cfa20b045a891eba2f7298afb8faec886d40d10b844830160b096fe99874') }}
-      SCCACHE_DIR: >-
-        ${{ (runner.os == 'Linux' && 'sccache-v0.4.1-x86_64-unknown-linux-musl')
-        || (runner.os == 'macOS' && 'sccache-v0.4.1-x86_64-apple-darwin')
-        || (runner.os == 'Windows' && 'sccache-v0.4.1-x86_64-pc-windows-msvc') }}
+      SHA_256_LINUX: f077d92ca86d71bc55aebeeb6e8dc557fef481446ccc82504aeedf1fe6e1f657 
+      SHA_256_LINUX_ARM: 263a43ba0cb211e5c1c10fe437c636d601bed7a47be0ca07beeba7973ba61461
+      SHA_256_MAC: a291f1d90c6b25726866f018ec6071fa4d20ca443ad91fe5dfb9740eb4ebc45a
+      SHA_256_WIN: 7508cfa20b045a891eba2f7298afb8faec886d40d10b844830160b096fe99874
       SCCACHE_BIN_FILE: sccache${{ runner.os == 'Windows' && '.exe' || '' }}
       SHASUM: ${{ runner.os == 'macOS' && 'shasum' || 'sha256sum' }}
     run: |
+      
+      case ${{ runner.os }} in
+        "Linux")
+          if [ "`uname -m`" == "x86_64" ]; then
```

Comment:
Can we use the `runner.arch` value instead of uname? It's usable across linux, windows, and macos, and we should do the same fork for macos too.

For example:

```yaml
      SCCACHE_FILE: >-
        ${{ (runner.os == 'Linux' && runner.arch == 'X64' && 'sccache-v0.8.2-x86_64-unknown-linux-musl.tar.gz')
        || (runner.os == 'Linux' && runner.arch == 'ARM64' && 'sccache-v0.8.2-aarch64-unknown-linux-musl.tar.gz')
        || (runner.os == 'macOS' && runner.arch == 'X64' && 'sccache-v0.8.2-x86_64-apple-darwin.tar.gz')
        || (runner.os == 'macOS' && runner.arch == 'ARM64' && 'sccache-v0.8.2-aarch64-apple-darwin.tar.gz')
        || (runner.os == 'Windows' && 'sccache-v0.8.2-x86_64-pc-windows-msvc.tar.gz') }}
```

Reply:
Was not aware of runner.arch was let me roll it back